test/context.test.js

```diff
@@ -0,0 +1,72 @@
+var loopback = require('loopback');
+var request = require('supertest');
+var Domain = require('domain');
+var assert = require('assert');
+var EventEmitter = require("events").EventEmitter;
+
+describe('check loopback.getCurrentContext', function() {
+  var app = loopback();
+  var runInOtherDomain;
+  var runnerInterval;
+
+  before(function(){
+    var called = false;
+    app.use(loopback.rest());
+    app.dataSource('db', { connector: 'memory' });
+
+    var TestModel = loopback.createModel({name: 'TestModel', properties: {name: String}, options: { base: 'Model'}});
+    app.model(TestModel, {dataSource: "db", public: true});
+
+    var emitterInOtherDomain = new EventEmitter();
+    Domain.create().add(emitterInOtherDomain);
+
+    runInOtherDomain = function(fn) {
+      emitterInOtherDomain.once('run', fn);
+    }
+
+    runnerInterval = setInterval(function() { emitterInOtherDomain.emit('run'); }, 10);
```

Comment:
I can't get what you described about "before/after hooks" here, could you give me more details ?